File: src/devtwin/services/redis.py
"""Redis service detector."""

from __future__ import annotations

from devtwin.services.base import ServiceDetector


class RedisDetector(ServiceDetector):
    name = "redis"
    default_port = 6379
    env_var_patterns = (r"^REDIS", r"^CACHE_URL$")
    dependency_names = ("redis", "ioredis", "django-redis")
    compose_image_hints = ("redis",)
    readme_keywords = ("redis",)

File: mcphub/devtwin/src/services/redis.ts
/** Redis service detector. */

import { ServiceDetector } from './base.js';

export class RedisDetector extends ServiceDetector {
  readonly name = 'redis';
  readonly defaultPort = 6379;
  override readonly envVarPatterns = ['^REDIS', '^CACHE_URL$'];
  override readonly dependencyNames = ['redis', 'ioredis', 'django-redis'];
  override readonly composeImageHints = ['redis'];
  override readonly readmeKeywords = ['redis'];
}
